### PR TITLE
Chai assert typing

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/chai/chai-tests.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import * as chai from 'chai';
 
 const expect = chai.expect;
-const assert = chai.assert;
+const assert: typeof chai.assert = chai.assert; // assertion aliases require explicit type annotation
```
Comment (joeykilpatrick) [inline types/chai/chai-tests.ts:5]: This change doesn't appear to do anything. With or without the additional annotation, the type of `assert` is still `Chai.AssertStatic`

File: types/chai/chai-tests.ts [hunk 2]
```diff
@@ -1405,7 +1405,10 @@ suite('assert', () => {
     test('assert', () => {
         const foo = 'bar' as string;
         assert(foo === 'bar', 'expected foo to equal `bar`');
+    });
 
+    test('assertFailure', () => {
+        const foo = 'bar' as string;
```
Comment (joeykilpatrick) [inline types/chai/chai-tests.ts:1411]: Not sure why this test case was split. This doesn't appear to be related to the PR.

**Review by joeykilpatrick — CHANGES_REQUESTED** [review]
Actual change looks good, just need to add a test case for it. Current changes to `chai-tests.ts` appear unrelated.

File: types/chai/chai-tests.ts (hunk 1 above)
Comment (joeykilpatrick) [reply types/chai/chai-tests.ts:5]: Nevermind, I understand this one now.

File: types/chai/chai-tests.ts (hunk 2 above)
Comment (jeremy-rifkin) [reply types/chai/chai-tests.ts]: This test case is split because of the new type information introduced by the assertion ``assert(foo === 'bar', ...);``:

```
Error: /home/felis/Projects/DefinitelyTyped/types/chai/chai-tests.ts:1408:7
ERROR: 1408:7   comment-format  comment must start with a space
ERROR: 1410:7   comment-format  comment must start with a space
ERROR: 1411:11  comment-format  comment must start with a space
ERROR: 1412:16  expect          TypeScript@4.7 compile error:
This condition will always return 'false' since the types '"bar"' and '"baz"' have no overlap.
```


I'm realizing there's a less disruptive solution than splitting the test case so I'm going to change it

**Comment by jeremy-rifkin** [comment]
Thanks so much for taking a look at the changes and for help with the test cases! Please let me know if further revision is needed

**Review by joeykilpatrick — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

**Comment by jeremy-rifkin** [comment]
Ready to merge
